pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -286,6 +288,14 @@ public void setCustomProperties(Map<String, String> properties) {
     _customProperties.putAll(properties);
   }
 
+  private void setDateTimeFormatSpec(DateTimeFormatSpec formatSpec) {
+    _dateTimeFormatSpec = formatSpec;
```

Comment:
We should also set `_timeColumnType` and `_segmentTimeUnit` accordingly

Reply:
I see `_segmentTimeUnit` has its own setter and getter, and it's only used for EPOCH time format.

Reply:
nvm, as we make setDateTimeFormatSpec public, setting the type and unit consistently is necessary.